src/main/java/net/oauth/jsontoken/JsonToken.java

```diff
@@ -82,11 +79,17 @@ public JsonToken(Signer signer) {
   public JsonToken(Signer signer, Clock clock) {
     Preconditions.checkNotNull(signer);
     Preconditions.checkNotNull(clock);
-    
+
+    header = new JsonObject();
+    header.addProperty(ALGORITHM_HEADER, signer.getSignatureAlgorithm().getNameForJson());
+    String keyId = signer.getKeyId();
+    if (keyId != null) {
+      header.addProperty(KEY_ID_HEADER, keyId);
+    }
```

Comment:
I prefer keeping the separate createHeader() method (even though it's only used once) because I think it makes the constructor easier to read.